rfcs/0006-gtfs-canonical-routes.md

```diff
@@ -0,0 +1,125 @@
+  created above. It should reference either a new (canonical) or previously-existing (regular
+  service) route pattern, depending on whether the trip's stopping pattern is currently served by
+  regular service. Each of these trips should be the `representative_trip_id` for their referenced
+  route patterns.
+- **stop_times.txt** includes all stops that trains in the route/direction passes through, even if
+  they are temporarily closed. Increment the arrival/departure times by one minute for each
+  subsequent stop. For a sample, see the attached GTFS file, the output of [✓ 🧪 🚝 Canonical
+  trips/patterns experiment](https://app.asana.com/0/881264583703207/1200210504369250), which
+  trialled this for Green Line D and Lowell Line (only in the outbound direction).
+
+This will be done as a new "step" of the Makefile called `canonical`, which runs between `truncate`
+and `current`. There will be a new a directory, `input/canonical/` to store any static data the step
+uses.
+
+The canonical step will be a Python module that reads in the `truncate` feed, and writes out its own
+feed after transforming it. The transformations are as above, and also will update any canonical
+trips to use current route patterns (and vice versa) if relevant, eliminating redundant route
+patterns.
+
+For Commuter Rail route patterns, the canonical one will specify every stop as non-flag, non-leave
+early stop. For both subway and Commuter Rail canonical trips, no data around bicycle allowances
+will be provided (`bikes_allowed=0`).
+
+# Drawbacks
+
+This is more complicated than not doing it.
+
+# Rationale and alternatives
+
+See proposed Concepts [1](https://github.com/mbta/technology-docs/pull/6#issuecomment-952315958), [4](https://github.com/mbta/technology-docs/pull/6#issuecomment-952315958), and [5](https://github.com/mbta/technology-docs/pull/6#issuecomment-962120434) in prior discussion.
+
+# Prior art
+
+Until now we've been doing ad-hoc adjustments to determine the route pattern of routes
+[here](https://github.com/mbta/api/blob/master/apps/state/config/config.exs#L145).
+
+# Unresolved questions
+
+Ensuring that we would not be in a situation where the sort order, perhaps ordered by
+frequency/proportion of trips, becomes a different order than desired for the main route pattern
+against the branch route pattern.
+
+How route patterns containing canonical representative trips are tagged or otherwise identified:
+Using a new `canonical_route_pattern` field in route_patterns.txt, or is the new
+`service_schedule_typicality` value sufficient?
+
+# Future possibilities
+
+We could expose this more clearly in the API. In the approach outlined here, the "canonical" route
+pattern is simply that with a typicality of `1`. In the alternative approach, the typicality would
+be `5`. That's not currently filterable in the API. In addition, the `service` can't be filtered by
+its typicality, most relevantly when included with `trips`.
```

Comment:
This seems to be partially referring to an earlier version of the proposal.

Reply:
Yeah, I not sure if I updated this section at all. I just tried to clean it up now.